### PR TITLE
js1w4 homework completed

### PR DESCRIPTION
**Pull request by sumanghimire79**

### CODE REVIEW

File: javascript/javascript1/week4/js1-w4.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+            console.log(result);
+        }
+        userName();
+    } else if (command.includes("What is your name") && theNameArray.length >= 1) {
+        function userName1() {
+            let sentence = "Your name is " + theNameArray[0];
+            console.log(sentence);
+        } userName1();
+    } else if (command == "" || theNameArray.length == 0) {
+        console.log("Your have mention your good name first");
```
Comment (esteficodes) [inline javascript/javascript1/week4/js1-w4.js:29]: Ok, but we can't really trust the user's input to be in the exact index that we plan, can we? What if the user writes something like "My name is Harold, good morning". It's better to use template literals so the function will always retrieve the name wherever it is located. ${name}

File: javascript/javascript1/week4/js1-w4.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    }
+    else if (command.includes("add fishing")) {
+        toDoList.unshift("fishing");
+        console.log("fishing added to your todo list");
+    } else if (command.includes("add singing shower")) {
+        toDoList.push("singing in shower");
+        console.log("Singing in the shower added to your todo list ");
```
Comment (esteficodes) [inline javascript/javascript1/week4/js1-w4.js:36]: Ok! And maybe there's a way to shorten this code and make it more efficient for every ${newTodo} that might be added. 🙂

File: javascript/javascript1/week4/js1-w4.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    } else if (command.includes("remove fishing")) {
+        toDoList.shift()
+        console.log('fishing removed from your todo list ');
+    } else if (command.includes("todo list")) {
+        console.log("your todo list has only " + toDoList.length + " activity. That means " + '"' + toDoList + '"' + " activity. !");
+    } else if (command.includes("today") || command.includes("date")) {
+        const monthNames = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
+        const date = new Date();
+        console.log("Todays date is : " + date.getDate() + " of " + monthNames[date.getMonth()] + ", " + date.getFullYear());
```
Comment (esteficodes) [inline javascript/javascript1/week4/js1-w4.js:45]: Template literals come handy in these cases 🙂

File: javascript/javascript1/week4/js1-w4.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+            console.log("The Answer is : " + (Number(calculateArray[0]) + Number(calculateArray[1])));
+        } else if (command.includes("*")) {
+            console.log("The Answer is : " + (Number(calculateArray[0]) * Number(calculateArray[1])));
+        } else if (command.includes("-")) {
+            console.log("The Answer is : " + (Number(calculateArray[0]) - Number(calculateArray[1])));
+        } else if (command.includes("/")) {
+            console.log("The Answer is : " + (Number(calculateArray[0]) / Number(calculateArray[1])));
+        } else if (command.includes("%")) {
+            console.log("The Answer is : " + (Number(calculateArray[0]) % Number(calculateArray[1])));
+        }
```
Comment (esteficodes) [inline javascript/javascript1/week4/js1-w4.js:60]: Ok! and what if we want to consider any number?

File: javascript/javascript1/week4/js1-w4.js [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    } else if (command.includes("set timer")) {
+        console.log('Timer is set for 5 seconds ! \n !! Do not blink the eyes !!');
+        function setTimer() {
+            console.log("5 second is up !");
+        }
+        setTimeout(setTimer, 5000,);
+    }
```
Comment (esteficodes) [inline javascript/javascript1/week4/js1-w4.js:67]: Nice! 👍

**Review by esteficodes — APPROVED** [review]
Good job! check out the comments and let me know if you have any questions. Whenever you are ready , you might merge your homework. 🙂

File: javascript/javascript1/week4/js1-w4.js (hunk 2 above)
Comment (sumanghimire79) [reply javascript/javascript1/week4/js1-w4.js]: i have tried

File: javascript/javascript1/week4/js1-w4.js (hunk 4 above)
Comment (sumanghimire79) [reply javascript/javascript1/week4/js1-w4.js:65]: .toFixed used. hope that is ok

File: javascript/javascript1/week4/js1-w4.js (hunk 1 above)
Comment (sumanghimire79) [reply javascript/javascript1/week4/js1-w4.js]: ok. managed to refactor it.

**Pull request merged**